lib/matplotlib/image.py

```diff
@@ -610,6 +613,23 @@ def set_data(self, A):
                 or self._A.ndim == 3 and self._A.shape[-1] in [3, 4]):
             raise TypeError("Invalid dimensions for image data")
 
+        if self._A.ndim == 3:
+            # If the input data has values outside the valid range (after
+            # normalisation), we issue a warning and then clip X to the bounds
+            # - otherwise casting wraps extreme values, hiding outliers and
+            # making reliable interpretation impossible.
+            high = 255 if np.issubdtype(self._A.dtype, np.integer) else 1
+            if self._A.min() < 0 or high < self._A.max():
```

Comment:
I wonder if we should pad this by an `eps` in case the 0-1 values came from some numerical computation, and we probably wouldn't want to spam the user with unnecessary warnings?

Reply:
We have to clip to exactly [0 .. 1], and for consistency I would prefer to always warn if clipping.  If users feel spammed, they can either fix their data or filter the logging output.

Reply:
I think the suggestion might have been to silently clip if (a-vmax)/(vmin - vmax) - 1 < 1e-10 or something that would just represent bit noise that would never affect the interpretation of the data.

Reply:
Yes, that's the question - I just think the answer is "No, we shouldn't".

Reply:
Agree that we shouldn't.